### DISCUSSION TITLE
Error: File 'BundleConfig.pb' was not found.

### DISCUSSION THREAD

**Discussion by P2b10** [airsdk/Adobe-Runtime-Support#1285] [Q&A]
Hi,

I compiled my android app with Adobe Animate 21.0.9 AIR SDK 33.1.1.633. I changed the .apk extension to .aab and uploaded to Google Play but I keep getting this error:

Error: File 'BundleConfig.pb' was not found.

Any advice?

**Comment by ajwfrost** [comment]
If you try opening that file (apk/aab) using a zip application such as 7-zip, you can see whether it's actually an .apk file or it's an .aab file because of the different structures. 

A bundle would look like:
![image](https://user-images.githubusercontent.com/6400804/138277019-41a37a85-7dad-4336-bd58-dda30d652dad.png)

An APK would look like:
![image](https://user-images.githubusercontent.com/6400804/138277105-1fe5d369-a4e1-45f8-9005-5347a42b6b1c.png)


If you're using Animate and you want to get it to output a bundle then you need to edit the adt.cfg file within the AIR SDK "lib" folder, and set:
`CreateAndroidAppBundle=true`
This way it will generate a bundle format, but within the .apk filename...

**Reply by P2b10** [reply]
Hi Andrew,

I followed your advice but now I got a folder named "AndroidStudioProject" that looks like this

![Screen Shot 2021-10-21 at 10 37 13](https://user-images.githubusercontent.com/49465984/138311438-68bd6131-ad3d-44f7-9278-8116d87605ba.png)

**Reply by P2b10** [reply]
What am I supposed to do with that?

Thanks!

**Reply by ajwfrost** [reply]
Hmm.. not sure we've tried things with both the `CreateAndroidAppBundle=true` setting and the `KeepAndroidStudioOutput=true` one. Does the root folder also contain a .aab file? If not, you can open that folder in Android Studio and create a bundle from there.. (or comment out `KeepAndroidStudioOutput` and re-create it from Animate?)

**Reply by P2b10** [reply]
There is no .aab file at all.

If I comment the "KeepAndroidStudioOutput" and publish with Animate, there is no folder, no .aab file nor an .apk file. Kind of weird.

This is how my adt.cfg file is configured
![Screen Shot 2021-10-21 at 12 25 31](https://user-images.githubusercontent.com/49465984/138327242-59a9ba97-589e-496a-8b68-ab51e94aca4e.png)
:

**Reply by ajwfrost** [reply]
Hmm... so with those settings, Animate should generate a file with an APK extension but that's in an App Bundle format. Can you uncomment the `DebugOut=true` line and try again, and then check the log file at `/Users/pablo/adt.log`?

I'm wondering whether there's a problem with the Java version being picked up there, but would expect Animate to report any errors like that..

thanks

**Reply by P2b10** [reply]
Hi Andrew,

Thanks for your reply.  I did what you said. When trying to compile now there is no folder, no .apk, no .aab.

The adt.log says “Invalid input: JAVA_HOME does not point to a valid JDK installation, which is required for Android App Bundle creation”.

I already broke my head trying to establish the JAVA_HOME in macOS Big Sur but it seems impossible. Any advice on how to fix that?

Thanks again.

Here is the complete adt.log:


*** ADT Logging at 2021/10/22 10:02:14 ***
Called by: /Applications/Adobe Animate 2021/AIR 33.1.1.633/lib/adt.jar
ADT 33.1.1.633 called with: -package -target apk-captive-runtime -arch armv8 -storetype PKCS12 -keystore /Users/pablo/Documents/PABLO/ACTUAL/APPS/0) CERTIFICADOS/CERTIFICADOS android/CERTIFICADO ANDROID/NUEVO - 2048/android2048.p12 -storepass xxxxxx /Users/pablo/Documents/PABLO/ACTUAL/APPS/DESARROLLO/PAGADAS/ANDROID/DOM 1/VERS 3 aab/1.swfs/9/BasicEarTrainingPRO.apk /Users/pablo/Documents/PABLO/ACTUAL/APPS/DESARROLLO/PAGADAS/ANDROID/DOM 1/VERS 3 aab/1.swfs/9/2-app.xml -C /Users/pablo/Documents/PABLO/ACTUAL/APPS/DESARROLLO/PAGADAS/ANDROID/DOM 1/VERS 3 aab/1.swfs/9/ 2.swf 192.png 144.png 96.png 72.png 48.png 36.png -C /Users/pablo/Documents/PABLO/ACTUAL/APPS/DESARROLLO/PAGADAS/ANDROID/DOM 1/VERS 3 aab/1.swfs/9/ lecciones
Overriding request for apk-captive-runtime, instead generating Android App Bundle
Ignoring architecture request for AAB target
AAB Packager entry point
Version in APK = 33.1.1.633, version of ADT = 33.1.1.633
Version in APK = 33.1.1.633, version of ADT = 33.1.1.633
Version in APK = 33.1.1.633, version of ADT = 33.1.1.633
Version in APK = 33.1.1.633, version of ADT = 33.1.1.633
..java.base/java.lang.Thread.getStackTrace(Unknown Source)
..com.adobe.air.DebugFile.printStack(DebugFile.java)
..com.adobe.air.InvalidInputException.<init>(InvalidInputException.java)
..com.adobe.air.apk.AABOutputStream.buildGradle(AABOutputStream.java)
..com.adobe.air.apk.AABPackager.renameOutput(AABPackager.java)
..com.adobe.air.ApplicationPackager.createPackage(ApplicationPackager.java)
..com.adobe.air.apk.AABPackager.createPackage(AABPackager.java)
..com.adobe.air.ADT.parseArgsAndGo(ADT.java)
..com.adobe.air.ADT.run(ADT.java)
..com.adobe.air.ADT.main(ADT.java)
Stack trace:
Invalid input: JAVA_HOME does not point to a valid JDK installation, which is required for Android App Bundle creation

**Reply by ajwfrost** [reply]
Ah, yes it looks like that isn't a JDK, it's a plug-in for running applets..? I thought they died out years ago...

Have you tried with the JAVA_HOME commented out?

On my Big Sur machine, there is no JAVA_HOME variable set, but 'javac' is a command in /usr/bin:
```
~ % which javac
/usr/bin/javac
~ % javac -version
javac 1.8.0_275
```

Although on another machine (Monterey on M1) I have:
```
~/Downloads/Android Studio.app/Contents/jre/Contents/Home/bin/javac
```
which is version 11, so if you have Android Studio then you could set that value to:
```
JAVA_HOME=/...location.../Android Studio.app/Contents/jre/Contents/Home
```
(I also did `which javac` on that machine and it's actually got an OpenJDK version downloaded via Homebrew, version 17 which I wouldn't recommend.. but it's still not actually got JAVA_HOME defined in the normal environment...)

Hope that helps!

thanks

**Comment by P2b10** [comment]
Hi Andrew,

Thanks for your reply. I installed jdk 15.0.2 and I solved the JAVA_HOME issue with this tutorial: https://www.youtube.com/watch?v=9nKIcK5-uxE

I have “the latest version of Android Studio an plug ins installed”.

I have macOS 11.6

But now I get this on my adt.log:


*** ADT Logging at 2021/10/22 12:50:17 ***
Called by: /Applications/Adobe Animate 2021/AIR 33.1.1.633/lib/adt.jar
ADT 33.1.1.633 called with: -version
*** ADT Logging at 2021/10/22 12:51:49 ***
Called by: /Applications/Adobe Animate 2021/AIR 33.1.1.633/lib/adt.jar
ADT 33.1.1.633 called with: -package -target apk-captive-runtime -arch armv8 -storetype PKCS12 -keystore /Users/pablo/Documents/PABLO/ACTUAL/APPS/0) CERTIFICADOS/CERTIFICADOS android/CERTIFICADO ANDROID/NUEVO - 2048/android2048.p12 -storepass xxxxxx /Users/pablo/Documents/PABLO/ACTUAL/APPS/DESARROLLO/PAGADAS/ANDROID/DOM 1/VERS 3 aab/1.swfs/9/BasicEarTrainingPRO.apk /Users/pablo/Documents/PABLO/ACTUAL/APPS/DESARROLLO/PAGADAS/ANDROID/DOM 1/VERS 3 aab/1.swfs/9/2-app.xml -C /Users/pablo/Documents/PABLO/ACTUAL/APPS/DESARROLLO/PAGADAS/ANDROID/DOM 1/VERS 3 aab/1.swfs/9/ 2.swf 192.png 144.png 96.png 72.png 48.png 36.png -C /Users/pablo/Documents/PABLO/ACTUAL/APPS/DESARROLLO/PAGADAS/ANDROID/DOM 1/VERS 3 aab/1.swfs/9/ lecciones
Overriding request for apk-captive-runtime, instead generating Android App Bundle
Ignoring architecture request for AAB target
AAB Packager entry point
Version in APK = 33.1.1.633, version of ADT = 33.1.1.633
Version in APK = 33.1.1.633, version of ADT = 33.1.1.633
Version in APK = 33.1.1.633, version of ADT = 33.1.1.633
Version in APK = 33.1.1.633, version of ADT = 33.1.1.633
Gradle build command:  /Library/Java/JavaVirtualMachines/jdk-15.0.2.jdk/Contents/Home/bin/java -Xmx1024m -Dorg.gradle.appname=gradlew -Dorg.gradle.jvmargs=-Xmx1024m -Duser.dir="/var/folders/pc/mrmwhj_s3hvb9rj5c3dxw5wh0000gn/T/76f27cf9-67e8-438a-a13c-21eadc36ab65" -cp "/Applications/Adobe Animate 2021/AIR 33.1.1.633/lib/android/bin/gradle-wrapper.jar":"/Applications/Adobe Animate 2021/AIR 33.1.1.633/lib/android/bin/gradle-cli.jar" org.gradle.wrapper.GradleWrapperMain :app:bundleRelease
..java.base/java.lang.Thread.getStackTrace(Unknown Source)
..com.adobe.air.DebugFile.printStack(DebugFile.java)
..com.adobe.air.ADTException.<init>(ADTException.java)
..com.adobe.air.apk.AABOutputStream.buildGradle(AABOutputStream.java)
..com.adobe.air.apk.AABPackager.renameOutput(AABPackager.java)
..com.adobe.air.ApplicationPackager.createPackage(ApplicationPackager.java)
..com.adobe.air.apk.AABPackager.createPackage(AABPackager.java)
..com.adobe.air.ADT.parseArgsAndGo(ADT.java)
..com.adobe.air.ADT.run(ADT.java)
..com.adobe.air.ADT.main(ADT.java)
Stack trace:
Unexpected failure: Unable to run java: com.adobe.air.ADTException: gradle tool failed: Warning: Mapping new ns http://schemas.android.com/repository/android/common/02 to old ns http://schemas.android.com/repository/android/common/01
Warning: Mapping new ns http://schemas.android.com/repository/android/generic/02 to old ns http://schemas.android.com/repository/android/generic/01
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/addon2/02 to old ns http://schemas.android.com/sdk/android/repo/addon2/01
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/repository2/02 to old ns http://schemas.android.com/sdk/android/repo/repository2/01
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/sys-img2/02 to old ns http://schemas.android.com/sdk/android/repo/sys-img2/01
Warning: Mapping new ns http://schemas.android.com/repository/android/common/02 to old ns http://schemas.android.com/repository/android/common/01
Warning: Mapping new ns http://schemas.android.com/repository/android/generic/02 to old ns http://schemas.android.com/repository/android/generic/01
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/addon2/02 to old ns http://schemas.android.com/sdk/android/repo/addon2/01
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/repository2/02 to old ns http://schemas.android.com/sdk/android/repo/repository2/01
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/sys-img2/02 to old ns http://schemas.android.com/sdk/android/repo/sys-img2/01

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:stripReleaseDebugSymbols'.
> NDK at /Users/pablo/Library/Android/sdk/ndk-bundle did not have a source.properties file

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 1m 4s

**Comment by P2b10** [comment]
Hi Andrew:
I think  fixed it. I downloaded the missing NDK on Android Studio and I have now the Bundle structure you mentioned earlier.

I will uploaded to Google Play and se what happens.

Thanks again for your help
![Screen Shot 2021-10-22 at 13 46 23](https://user-images.githubusercontent.com/49465984/138507657-8a88672d-e3ff-477a-b418-61388a41d8c7.png)
!

**Reply by P2b10** [reply]
I did an internal test on Google Play and I was able to download the app to my android device.
So far my app is in review on Google Play.

Seems OK.

**Reply by ajwfrost** [reply]
That's great .. although I have no idea why it was trying to get you to install the NDK..?! You shouldn't need that at all for building an AIR application..!

**Reply by P2b10** [reply]
Such is life in the tropics my friend! 

I will let you know when it goes through Google Play´s review.

Best!
